### PR TITLE
feat: Add prcreation == approval

### LINKED ISSUE #3928: New option prCreation=approval
**What would you like Renovate to be able to do?**

Create branches for testing but do not create PRs unless they are approved in the master issue.

**Describe the solution you'd like**

I think there should be a new heading/category for updates that are (a) created as a branch, but (b) awaiting approval for PR creation.

Essentially, masterIssueApproval=true will (still) mean approval needed before creating branches, while prCreation=approval will mean approval needed before creating PRs.

### PR DESCRIPTION
**Pull request by praveshtora**
* In pr worker checking for ` config.prCreation === 'approval' && !existingPr && masterIssueCheck !== 'approvePr'` and if true, flagging the branch.
* After this masterIssueApproval flag will create list of branches that have not been created yet while prCreation== approval will create list of branches which have been created but not raised a PR.

Closes #3928

### CODE REVIEW

**Comment by rarkins** [comment]
We’ll need to rename/rephrase the master issue descriptions as we now have the possibility of both branch approval and PR approval as shown in your screenshot. I’ll work on those last

**Comment by praveshtora** [comment]
@rarkins I have tested it manually, by setting masterIssueApproval=true and prCreation=approval. After that when I checked a branch in master Issue approval, I can see branches being created and no PRs of that branch and the branch can be seen in the PR approval section. Any other particular test case do I need to look into?

**Comment by rarkins** [comment]
That sounds like a good start. When you tick the branch approval, does it open up the PR or stop at the branch? I would guess it creates the PR too (ie skips PR approval step)

**Comment by praveshtora** [comment]
It stops at the branch. There is no PR. I am handling it in PR worker.

**Comment by rarkins** [comment]
I would have expected that the PR worker thinks it's approved in master issue and goes ahead. How does it differentiate between a branch approval tick and a PR approval tick?

**Comment by praveshtora** [comment]
It differentiates by the ``type``. In PR approval tick, the type is `approvePr` while in branch approval it just `approve`. While parsing the `issue` body, we use regex to get the type and then in pr worker I am matching it with this type to ensure PR.

**Comment by rarkins** [comment]
Oh, great. I guess I forgot that code :D

**Comment by praveshtora** [comment]
Are there any edge cases that I need to consider?

**Comment by rarkins** [comment]
Looks good to me for now. I’ll test and try to think of any edge cases before merging.

**Comment by rarkins** [comment]
@praveshtora I've tested it and made some edits to your branch. Can you update tests to get it passing on CI?

File: lib/workers/pr/index.js
Comment (rarkins) [inline lib/workers/pr/index.js]: Let's make this 'needs-pr-approval' instead of 'needs-approval' from the very start so that we don't have to remap it.

**Review by rarkins — CHANGES_REQUESTED** [review]
Rename 'needs-approval' to 'needs-pr-approval', and update or remove screenshot in PR description as it's currently old/incorrect.

**Pull request merged**
